src/main/java/com/aws/greengrass/iot/IotCloudHelper.java

```diff
@@ -74,9 +82,10 @@ public IotCloudResponse sendHttpRequest(final IotConnectionManager connManager,
             innerRequestBuilder.appendHeader(HTTP_HEADER_THING_NAME, thingName);
         }
 
-        ExecutableHttpRequest request = connManager.getClient().prepareRequest(HttpExecuteRequest.builder()
-                .contentStreamProvider(body == null ? null : () -> new ByteArrayInputStream(body))
-                .request(innerRequestBuilder.build()).build());
+        ExecutableHttpRequest request = connManager.getClient()
+                .prepareRequest(HttpExecuteRequest.builder()
+                        .contentStreamProvider(body == null ? null : () -> new ByteArrayInputStream(body))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The code is not properly closing resources, which can lead to resource leaks. Unclosed resources may cause memory leaks or other system resource issues, potentially degrading application performance over time. If an exception occurs before the explicit close() call, the resource will not be closed, increasing the risk of leaks. To remediate this, use the try-with-resources statement to automatically close AutoCloseable resources, ensuring proper resource management even when exceptions occur.

[Learn more](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)